reana-ui/src/components/Table.js

```diff
@@ -0,0 +1,221 @@
+/*
+	-*- coding: utf-8 -*-
+
+	This file is part of REANA.
+	Copyright (C) 2018 CERN.
+
+	REANA is free software; you can redistribute it and/or modify it under the
+	terms of the GNU General Public License as published by the Free Software
+	Foundation; either version 2 of the License, or (at your option) any later
+	version.
+
+	REANA is distributed in the hope that it will be useful, but WITHOUT ANY
+	WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+	A PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+	You should have received a copy of the GNU General Public License
+	along with REANA; if not, see <http://www.gnu.org/licenses>.
+
+	In applying this license, CERN does not waive the privileges and immunities
+	granted to it by virtue of its status as an Intergovernmental Organization or
+	submit itself to any jurisdiction.
+*/
+
+import React, { Component } from "react";
+import { Icon, Menu, Segment, Table } from "semantic-ui-react";
+import TableProgress from "./TableProgress";
+import TableActions from "./TableActions";
+import _ from "lodash";
+
+const URL = "http://reana-dev.cern.ch/api/workflows?";
+const TOKEN = "yCmoYC08hwqp7hFhpgwUUS6fIG8OflCvakCUKDi5cVE";
+const POOLING_PERIOD = 5;
+
+export default class SortTable extends Component {
+  /**
+   * Variables defining the state of the table
+   */
+  state = {
+    column: null,
+    data: [],
+    direction: null,
+    interval: null
+  };
+
+  /**
+   * Transforms millisecond into a 'HH MM SS' string format
+   */
+  static msToTime(millis) {
+    let seconds = Math.floor((millis / 1000) % 60);
```

Comment:
what about using the Date object? something like
```JavaScript
let string_date = "2018-07-30T15:52:44";
let date_obj = new Date(string_date)
let seconds = date_obj.getSeconds()
let minutes = date_obj.getMinutes()
let hours = date_obj.getHours()
```

Reply:
That way of operating is used to plot the creation date. However, for the `Duration` field, we should compare every date entity (secs, mins, hours, days, months, year).

Because of this reason, I transform the date into a UNIX stamp number, and compute the difference between `Date.now()` and that number.